api/proto/teleport/legacy/types/types.proto

```diff
@@ -1857,13 +1857,9 @@ message DeviceTrust {
   // Defaults to "optional" for Enterprise.
   string Mode = 1 [(gogoproto.jsontag) = "mode,omitempty"];
 
-  // Enable device auto-enroll.
-  // Auto-enroll lets any user issue a device enrollment token for a known
-  // device that is not already enrolled.
-  // `tsh` takes advantage of auto-enroll to automatically enroll devices on
-  // user login, when appropriate.
-  // The effective cluster Mode still applies: AutoEnroll=true is meaningless if
-  // Mode="off".
+  // Use AutoEnrollMode instead.
+  // AutoEnroll is completely ignored in favor of AutoEnrollMode.
```

Comment:
This comment doesn't seem to be completely true. If AutoEnroll == true and AutoEnrollMode == disabled then `AuthPreferenceV2.CheckAndSetDefaults()` will return an error.

Reply:
I think that's alright? In most cases it is ignored and the error explains itself.

Reply:
Adding to the reply, I don't want to explain too many caveats. The error could change or be dropped, but the main reason for the comment remains: do not use or rely on the old flag.